### PR TITLE
fix(canvas): invalidate canvas on finish layer

### LINKED ISSUE #6005: lvgl 9.1 canvas image not updated on the screen, unless canvas invalidated
### LVGL version

v9.1.0

### What happened?

I cannot update the lines on canvas "in live" unless the canvas or the main object are invalidated.
The lines are drawn on the buffer, but the buffer is not updated to the screen.

### How to reproduce?

I pretty much follow example of canvas, trying to draw lines on the canvas.
I create a canvas as part of my own lv_obj. 
Then I create a static draw buffer and set it.
To draw some lines I init a layer, draw the lines, and then finish the layer.

```
LV_DRAW_BUF_DEFINE(canvas_buf, 480, 480, LV_COLOR_FORMAT_ARGB8888);

my_obj->canvas = lv_canvas_create((lv_obj_t*)my_obj);
lv_canvas_set_draw_buf(my_obj->canvas, &canvas_buf);

lv_obj_align(my_obj->canvas, LV_ALIGN_CENTER, 0, 0);
lv_canvas_fill_bg(my_obj->canvas, lv_color_black(), LV_OPA_COVER);

lv_canvas_init_layer(my_obj->canvas, &layer);

// create and init line_dsc
for (uint32_t tick = 0; tick < 100; tick++) {
    lv_draw_line(&layer, &line_dsc);
}

lv_canvas_finish_layer(my_obj->canvas, &layer);
```

Unless I do not `lv_obj_invalidate(canvas)` the lines are not shown.
I can also see the lines after I change screen with `lv_screen_load(other_obj)` to another object and then get back `lv_screen_load(my_obj)`.

### PR DESCRIPTION
**Pull request by liamHowatt**
### Description of the feature or fix

Fixes #6005

`lv_canvas_finish_layer` does not invalidate the canvas. Changes are not drawn unless other canvas functions are called before the refresh.

Invalidate the canvas in `lv_canvas_finish_layer`. Add a test case that asserts all canvas drawing functions invalidate the canvas.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: tests/src/test_cases/widgets/test_canvas.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    lv_canvas_set_px(canvas, 0, 0, lv_color_black(), LV_OPA_COVER);
+    lv_refr_now(NULL);
+    TEST_ASSERT(draw_counter == 2);
+
+    lv_canvas_fill_bg(canvas, lv_color_black(), LV_OPA_COVER);
+    lv_refr_now(NULL);
+    TEST_ASSERT(draw_counter == 3);
+
+    {
```
Comment (kisvegabor) [inline tests/src/test_cases/widgets/test_canvas.c]: Why the `{` is needed?

Comment (liamHowatt) [reply tests/src/test_cases/widgets/test_canvas.c]: Removed. It's not needed. I just wanted to limit the scope of `layer` and `line_dsc`.

File: src/widgets/canvas/lv_canvas.c [hunk 2]
```diff
@@ -336,6 +336,7 @@ void lv_canvas_finish_layer(lv_obj_t * canvas, lv_layer_t * layer)
     while(layer->draw_task_head) {
         lv_draw_dispatch_wait_for_request();
         lv_draw_dispatch_layer(lv_obj_get_display(canvas), layer);
+        lv_obj_invalidate(canvas);
     }
```
Comment (FASTSHIFT) [inline src/widgets/canvas/lv_canvas.c:340]: How about writing it this way? Reduce the number of `lv_obj_invalidate` calls:
```c
void lv_canvas_finish_layer(lv_obj_t * canvas, lv_layer_t * layer)
{
    if(!layer->draw_task_head) {
        return;
    }
    while(layer->draw_task_head) {
        lv_draw_dispatch_wait_for_request();
        lv_draw_dispatch_layer(lv_obj_get_display(canvas), layer);
    }
    lv_obj_invalidate(canvas);
}
```

Comment (liamHowatt) [reply src/widgets/canvas/lv_canvas.c:340]: Brilliant! This is why we do reviews.

**Comment by kisvegabor** [comment]
@liamHowatt I think PR was closed by accident so I reopened it.

**Review by kisvegabor — APPROVED** [review]

**Comment by liamHowatt** [comment]
> @liamHowatt I think PR was closed by accident so I reopened it.

Yes, thank you! I remember what caused it.

**Review by FASTSHIFT — APPROVED** [review]

**Pull request merged**